Zend/zend_compile.c

```diff
@@ -6453,6 +6467,10 @@ void zend_compile_class_decl(zend_ast *ast) /* {{{ */
 	}
 
 	CG(active_class_entry) = ce;
+	if (extends_ast) {
+		zend_string *extends_name = zend_ast_get_str(extends_ast);
+		CG(active_class_entry_parent_name) = zend_string_copy(extends_name);
```

Comment:
This assigns the non-resolved class name. Either this should assign the correct class name, or should just be a boolean for whether it has a parent, to prevent confusion when other code wants to use this member in the future.

Reply:
I think other code will need the name of the parent; for instance we reject `string $parent = parent::class` in parameters but there's no need to actually do this now. Will investigate resolving the class name.

Reply:
Just to check, what do you mean by "non-resolved"? I assume `Bar` in `use Foo as Bar; class Baz extends Bar {}` or something?

Reply:
Yes, that's right. You can either use zend_resolve_class_name, or try to get the already resolved name from above, though extracting it from literals might be a bit inconvenient.

Reply:
Dmitry has changed this code recently; it should be easy to use now.